api/envoy/config/core/v3/protocol.proto

```diff
@@ -389,3 +389,8 @@ message GrpcProtocolOptions {
 
   Http2ProtocolOptions http2_protocol_options = 1;
 }
+
+// [#not-implemented-hide:]
+message Http3ProtocolOptions {
+  Http2ProtocolOptions http2_protocol_options = 1;
```

Comment:
How does this configuration address the difference between H2 and H3? From a quick glance at the current spec, there are differences around SETTINGS, WINDOW_UPDATE frames.

Reply:
I guess the options are
1) using Http2ProtocolOptions, and config-validating that fields which don't make sense don't apply
2) copying most of the fields out of Http2ProtocolOptions so we don' t have the fields which make sense
3) splitting Http2ProtocolOptions so we have http2/http3/shared today

I think 3) is undesirable today, especially as http3 hasn't fully landed yet.  If we agree it may be the end goal, I think 1 is the least churn, so what I think we should go with, but I'm not committed to it so if you see other ideas or have other pros/cons let me know

Reply:
Yeah agreed 1 is least churn, can we add some comment to address those fields?